cats/__init__.py

```diff
@@ -4,21 +4,50 @@
 import yaml
 import sys
 
-from .check_clean_arguments import validate_jobinfo, validate_duration
+from .check_clean_arguments import validate_jobinfo, validate_duration, validate_location
 from .optimise_starttime import get_starttime  # noqa: F401
 from .CI_api_interface import API_interfaces
 from .CI_api_query import get_CI_forecast  # noqa: F401
 from .parsedata import avg_carbon_intensity  # noqa: F401
 from .carbonFootprint import greenAlgorithmsCalculator
 
 def parse_arguments():
+    """
+    Parse command line arguments
+    :return: [dict] parsed arguments
+    """
     parser = ArgumentParser(prog="cats", description="A climate aware job scheduler")
 
-    #parser.add_argument("program")
-    parser.add_argument("--loc")
-    parser.add_argument("-d", "--duration", type=int, required=True)
-    parser.add_argument("--jobinfo")
-    parser.add_argument("--config")
+    ### Required
+
+    parser.add_argument("-d", "--duration", type=int, required=True, help="Expected duration of the job in minutes.")
+
+    ### Optional
+
+    parser.add_argument(
+        "--api-carbonintensity", type=str,
+        help="[optional] which API should be used to obtain carbon intensity forecasts. Overrides `config.yml`."
+             "For now, only choice is 'carbonintensity.org.uk' (UK only) (default: 'carbonintensity.org.uk')"
```

Comment:
Any argument starting with `--` will be considered optional and this should be automatically reflected in the generated help. You can also specify the default value within the call to `add_argument`:
```
parser.add_argument("--api=carbonintensity", type=str, default="carbonintensity.org.uk")
```
that will save you some tests down below.

Reply:
Good call regarding the default value in general, but here we want the command line to override the config file only if the command line is provided (and default value is only used if neither command line nor config file is specified). If we use a `default` value in `add_argument`, we would lose that information I think? Or would you know a cleaner way to do it?

Reply:
You're right that all the `-` or `--` arguments are considered optional by convention, I'll remove the `[optional]` in the description. Looking at the [SO answer here](https://stackoverflow.com/questions/24180527/argparse-required-arguments-listed-under-optional-arguments), we should highlight that `-d` is required (but it shows under "options" for now, so if anyone has recommendations on the best way to deal with it? (there are some suggestions on the SO thread)

Reply:
>  Here we want the command line to override the config file only if the command line is provided [...]. If we use a default value in add_argument, we would lose that information I think?

Hmm! Yes that's tricky. I don't see a neater way now.. Maybe set the default value based on the the content (or non-content) of the config file? Maybe not worth the effort at this point.